crates/cext/src/transpiler/target.rs

```diff
@@ -0,0 +1,741 @@
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_property_map_length(property_map: *const PropertyMap) -> usize {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let prop_map = unsafe { const_ptr_as_ref(property_map) };
+    prop_map.0.len()
+}
+
+/// @ingroup QkPropsMap
+/// Frees the property map.
+///
+/// @param property_map The pointer to the mapping object to be freed.
+///
+/// # Example
+///
+///     QkPropsMap *props_map = qk_property_map_new();
+///     qk_property_map_free(props_map);
+///
+/// # Safety
+///
+/// The behavior is undefined if ``property_map`` is not a valid,
+/// non-null pointer to a ``QkPropsMap`` object.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_property_map_free(property_map: *mut PropertyMap) {
+    if !property_map.is_null() {
+        if !property_map.is_aligned() {
+            panic!("Attempted to free a non-aligned pointer.")
+        }
+
+        // SAFETY: We have verified the pointer is non-null and aligned, so it should be
+        // readable by Box.
+        unsafe {
+            let _ = Box::from_raw(property_map);
+        }
+    }
+}
+
+/// @ingroup QkPropsMap
+/// Adds an InstructionProperty based on its assigned qargs.
+///
+/// @param property_map The pointer to the mapping object.
+/// @param qargs A pointer to the array of ``uint32_t`` qubit indices to add the
+///     gate on, can be a null pointer to check for global properties.
+/// @param num_qubits The length of the qargs array.
+/// @param duration The instruction's duration in seconds on the specific set of
+///     qubits.
+/// @param error The instruction's average error rate on the specific set of qubits.
+///
+/// # Example
+///
+///     QkPropsMap *props_map = qk_property_map_new();
+///     uint32_t qargs[2] = {0, 1};
+///     qk_property_map_add(props_map, qargs, 2, 0.0, 0.1);
+///
+/// # Safety
+///
+/// The behavior is undefined if ``property_map`` is not a valid, non-null pointer
+/// to a ``QkPropsMap`` object.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_property_map_add(
+    property_map: *mut PropertyMap,
+    qargs: *mut u32,
+    num_qubits: u32,
+    duration: f64,
+    error: f64,
+) {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let prop_map = unsafe { mut_ptr_as_ref(property_map) };
+    // SAFETY: Per the documentation the qubits pointer is an array of num_qubits elements
+    let qubits: Qargs = unsafe { parse_qargs(qargs, num_qubits) };
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let duration = if duration.is_nan() {
+        None
+    } else {
+        Some(duration)
+    };
+    let error = if error.is_nan() { None } else { Some(error) };
+    prop_map
+        .0
+        .insert(qubits, Some(InstructionProperties::new(duration, error)));
+}
+
+/// @ingroup QkTarget
+/// Adds a non-parameteric StandardGate to the ``Target``.
```

Comment:
The same comment also in some other locations below: standard gates are exposed as the enum ``QkGate``
```suggestion
/// Adds a non-parameteric ``QkGate`` to the ``Target``.
```

Reply:
Renamed in 395a194